### ISSUE TITLE
DateTime.timestamp_nanos() on valid DateTime panics instead of returning a result/option

### ISSUE THREAD

**Issue by 0x5c** [chronotope/chrono#586]
While it is documented behaviour, a valid DateTime should not panic while getting a timestamp from it.

### Minimal repro

```rust
let naive = NaiveDateTime::from_timestamp_opt(456456456456, 0).unwrap(); //Does not panic
let datetime = DateTime::<Utc>::from_utc(naive, Utc)); //Still no panic
println!("{}", datetime.timestamp_nanos()); //This panics

```

---

### Potential low-effort fix

```rust
pub fn timestamp_nanos(&self) -> Option<i64> {
    let as_ns = self.timestamp().checked_mul(1_000_000_000);
    match as_ns {
        Some(ns) => Some(ns + i64::from(self.timestamp_subsec_nanos())),
        None => None,
    }
}
```

While testing this potential fix, I uncovered the following points where other unrelated operations on DateTimes call `timestamp_nanos()`, which suggests to me that this might of greater concern than something that can be chalked up as a low importance "formatting bug".

`round.rs: 157`  
`round.rs: 186`

An alternate fix would be to add checked variants of `timestamp_nanos()` and `timestamp_millis()`.

---

As for my use-case, I can easily work around the issue, but this is still a big problem for other use-cases.

**Comment by 0x5c** [comment]
@Milo123459 Finally got to test this patch, and it works perfectly for me

**Comment by Milo123459** [comment]
Great! I'll fix that PR in a moment

**Comment by mlegner** [comment]
This seems to be addressed by #1275.

**Comment by 0x5c** [comment]
I think this can be closed then
